### PR TITLE
fixed the scan bug

### PR DESCRIPTION
**Pull request by tigershi**

### CODE REVIEW

File: g11n-ws/modules/md-service-i18n-l3/src/main/java/com/vmware/vip/core/messages/utils/LocaleUtility.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ public static boolean isDefaultLocale(Locale locale) {
 	 */
 	public static boolean isDefaultLocale(String locale) {
 		boolean isDefault = false;
-		if (locale != null && locale != "") {
+		if (locale != null && !locale.equals("")) {
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-service-i18n-l3/src/main/java/com/vmware/vip/core/messages/utils/LocaleUtility.java]: Change to StringUtil.IsEmpty()

File: g11n-ws/tools/tool-trans-fetcher/src/main/java/com/vmware/vip/fetcher/common/FileUtils/OrderedPropertiesUtils.java [hunk 2]
```diff
@@ -37,8 +37,8 @@ public static OrderedProperties loadFromFile(String fileName) {
         OrderedProperties p = new OrderedProperties();
 
         try {
-            InputStream inputStream = new FileInputStream(fileName);
-            p.load(new InputStreamReader(inputStream, ConstantsUnicode.UTF8));
+          
+            p.load(new InputStreamReader(new FileInputStream(fileName), ConstantsUnicode.UTF8));
```
Comment (Xiaochao8) [inline g11n-ws/tools/tool-trans-fetcher/src/main/java/com/vmware/vip/fetcher/common/FileUtils/OrderedPropertiesUtils.java]: Where to close the stream?

File: g11n-ws/tools/tool-trans-fetcher/src/main/java/com/vmware/vip/fetcher/monitor/UpdateThread.java [hunk 3]
```diff
@@ -12,17 +12,17 @@
...
 public class UpdateThread extends Thread {
-
-    @SuppressWarnings("static-access")
+    private static boolean flag = true;
     synchronized public void run() {
         MultiComponentsDTO syncObject = new MultiComponentsDTO();
         synchronized (syncObject) {
-            boolean flag = true;
+           
             while (flag) {
```
Comment (Xiaochao8) [inline g11n-ws/tools/tool-trans-fetcher/src/main/java/com/vmware/vip/fetcher/monitor/UpdateThread.java]: use 'true' directly here.

File: g11n-ws/vip-common/src/main/java/com/vmware/vip/common/constants/ConstantsMsg.java [hunk 4]
```diff
@@ -25,6 +25,7 @@
          "The pseudo %s, return the received source with pseudo tag.";
    public static final String PSEUDO_FOUND = "The pseudo %s.";
    public static final String PARAM_NOT_VALIDATE = "The parameter is not pass the validate";
+   public static final String FIFE_NOT_FOUND= "File is not existing: ";
```
Comment (Xiaochao8) [inline g11n-ws/vip-common/src/main/java/com/vmware/vip/common/constants/ConstantsMsg.java]: This isn't related to scan errors

File: g11n-ws/vip-common/src/main/java/com/vmware/vip/common/i18n/status/APIResponseStatus.java [hunk 5]
```diff
@@ -15,6 +15,7 @@
     public static final Response NO_CONTENT = new Response(204, "No Content");
     public static final Response TRANSLATION_NOT_READY = new Response(205, "Collected resource's translations is not ready");
     public static final Response TRANSLATION_READY = new Response(206, "Collected resource's translations have been ready");
+    public static final Response MULTTRANSLATION_PART_CONTENT = new Response(207, "Part of translations is available");
```
Comment (Xiaochao8) [inline g11n-ws/vip-common/src/main/java/com/vmware/vip/common/i18n/status/APIResponseStatus.java]: This isn't related to scan errors

File: g11n-ws/vip-common/src/main/java/com/vmware/vip/common/utils/LocaleUtils.java [hunk 6]
```diff
@@ -85,7 +85,7 @@ public static boolean isDefaultLocale(Locale locale) {
 	 */
 	public static boolean isDefaultLocale(String locale) {
 		boolean isDefault = false;
-		if (locale != null && locale != "") {
+		if (locale != null && !locale.equals("")) {
```
Comment (Xiaochao8) [inline g11n-ws/vip-common/src/main/java/com/vmware/vip/common/utils/LocaleUtils.java]: StringUtil

File: g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/HttpRequester.java [hunk 7]
```diff
@@ -170,10 +170,10 @@ public String getBaseURL() {
 	 * @return
 	 */
 	public static String post(String url, Map<String, String> params,Map<String, String> headers) {
-		HttpURLConnection conn = HttpRequester.createConnection(url);
 		BufferedInputStream bis = null;
 		ByteArrayOutputStream bos = null;
 		try {
+		    HttpURLConnection conn = HttpRequester.createConnection(url);
```
Comment (Xiaochao8) [inline g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/HttpRequester.java]: Where to close conn?

File: g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/HttpRequester.java [hunk 8]
```diff
@@ -230,11 +230,11 @@ public static String post(String url, Map<String, String> params,Map<String, Str
 	public static String sendPostJson(String url, Map<String, String> params,Map<String, String> headers) {
 		
 		
-		HttpURLConnection conn = HttpRequester.createConnection(url);;
       
 		BufferedInputStream bis = null;
 		ByteArrayOutputStream bos = null;
 		try {
+		HttpURLConnection conn = HttpRequester.createConnection(url);;
```
Comment (Xiaochao8) [inline g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/HttpRequester.java]: closed?

File: g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/HttpRequester.java [hunk 9]
```diff
@@ -385,6 +386,7 @@ public boolean verify(String hostname, SSLSession session) {
 				httpsConn.setSSLSocketFactory(getSocketFactory());
 			}
 		} catch (IOException e) {
+		   throw e;
```
Comment (Xiaochao8) [inline g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/HttpRequester.java]: Don't try this exception.

File: g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/VipRSAUtils.java [hunk 10]
```diff
@@ -100,13 +104,17 @@ public static String rsaEncrypt(String encryptStr, String pubKeyStr) {
...
-				cipher.init(Cipher.ENCRYPT_MODE, publicKey);
-			} catch (InvalidKeyException e1) {
-				// TODO Auto-generated catch block
-				e1.printStackTrace();
+			if(cipher != null) {
+			    try {
+	                cipher.init(Cipher.ENCRYPT_MODE, publicKey);
+	            } catch (InvalidKeyException e1) {
+	                // TODO Auto-generated catch block
+	                e1.printStackTrace();
```
Comment (Xiaochao8) [inline g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/VipRSAUtils.java]: Please use logger

File: g11n-ws/tools/tool-trans-fetcher/src/main/java/com/vmware/vip/fetcher/monitor/UpdateThread.java (hunk 3 above)
Comment (tigershi) [reply g11n-ws/tools/tool-trans-fetcher/src/main/java/com/vmware/vip/fetcher/monitor/UpdateThread.java]: use the 'true' directly is not recommend in java program

File: g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/HttpRequester.java (hunk 7 above)
Comment (tigershi) [reply g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/HttpRequester.java]: conn no close method

File: g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/HttpRequester.java (hunk 9 above)
Comment (tigershi) [reply g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/HttpRequester.java]: this is private method, why don't show this exception

File: g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/VipRSAUtils.java (hunk 10 above)
Comment (tigershi) [reply g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/utils/VipRSAUtils.java]: this is a util class, and it's a history leave issue

File: g11n-ws/tools/tool-trans-fetcher/src/main/java/com/vmware/vip/fetcher/monitor/UpdateThread.java (hunk 3 above)
Comment (Xiaochao8) [reply g11n-ws/tools/tool-trans-fetcher/src/main/java/com/vmware/vip/fetcher/monitor/UpdateThread.java]: flag is unnecessary

Comment (tigershi) [reply g11n-ws/tools/tool-trans-fetcher/src/main/java/com/vmware/vip/fetcher/monitor/UpdateThread.java]: no, use the multi-thread, not recommend  use direct 'true' in run method

File: g11n-ws/tools/tool-trans-fetcher/src/main/java/com/vmware/vip/fetcher/common/FileUtils/OrderedPropertiesUtils.java (hunk 2 above)
Comment (tigershi) [reply g11n-ws/tools/tool-trans-fetcher/src/main/java/com/vmware/vip/fetcher/common/FileUtils/OrderedPropertiesUtils.java]: this doesn't need to close, it is a parameter not variable

**Pull request merged**